### ISSUE TITLE
KVO NSKeyValueObservingOptionOld value invalid when observing multiple keypaths on an object.

### ISSUE THREAD

**Issue by rupertdaniel** [Microsoft/WinObjC#2709]
If an observer is observing multiple keypaths on the same object, the value for NSKeyValueObservingOptionOld can be invalid for some of the change notifications for a keypath.

This can be reproduced using the following example:

```
@interface TestObject
@property (nonatomic, strong) NSString *v;
@end

@implementation TestObject

- (instancetype)init
{
	if (self = [super init]) {
		self.v = [[NSProcessInfo processInfo] globallyUniqueString];
	}

	return self;
}

@end

@interface Controller : NSObject

@property TestObject *testObject;

@end

@implementation

- (instancetype)init
{
  if (self = [super init]) {
    [self addObserver:self forKeyPath : @"testObject" options : NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionNew | NSKeyValueObservingOptionOld context : NULL];
    [self addObserver:self forKeyPath : @"testObject.v" options : NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionNew | NSKeyValueObservingOptionOld context : NULL];
  }

  return self;
}

- (void)doSomething
{
  self.testObject = [[TestObject alloc] init];
}

-(void)observeValueForKeyPath:(NSString *)keyPath ofObject : (id)object change : (NSDictionary *)change context : (void *)context
{
	id oldValue = [change objectForKey : NSKeyValueChangeOldKey];
	id newValue = [change objectForKey : NSKeyValueChangeNewKey];

	NSLog(@"%@ old %@ new %@", keyPath, oldValue, newValue);
}

@end
```

When run and `[Controller doSomething]` is called the change logging will be...
```
testObject.v old 49A2D6B0-0C0E-D54E-821E-47145BF2E0B7 new 5871790A-36D0-6748-B41C-A8BFB017ACA0
testObject old 49A2D6B0-0C0E-D54E-821E-47145BF2E0B7 new <TestObject: 0ED63AB4>
```

As you can see the old object value for the keyPath ```testObject``` is actually the old value for the ```testObject.v``` keyPath.

**Comment by rajsesh-msft** [comment]
@DHowett-MSFT says @aballway already fixed this in a PR in flight.

**Comment by rajsesh-msft** [comment]
Fixed with #2693
